In [1]:
%matplotlib inline

In [2]:
test = pd.read_csv('C:\\dss\\team_project2\\walmart\\test.csv')
test

VisitNumber Weekday           Upc  ScanCount  \
0                 1  Friday  7.250339e+10          1   
1                 1  Friday  1.707711e+09          1   
2                 1  Friday  8.947000e+10          1   
3                 1  Friday  8.849121e+10          1   
4                 2  Friday  2.840015e+09          1   
5                 2  Friday  7.874205e+09          1   
6                 2  Friday  8.745860e+10          1   
7                 2  Friday  8.745860e+10          1   
8                 3  Friday  7.410811e+09          1   
9                 3  Friday  7.410811e+09         -1   
10                4  Friday  7.287926e+09          1   
11                6  Friday  7.603139e+09          1   
12                6  Friday  7.603139e+09         -1   
13               13  Friday  4.470036e+09          1   
14               13  Friday  5.200033e+09          1   
15               14  Friday  4.282561e+09          1   
16               14  Friday  6.454167e+10          1   
17               14  Friday  2.840000e+09          1   
18               14  Friday  3.400000e+09          2   
19               14  Friday  2.340035e+09          1   
20               14  Friday  7.800000e+09          1   
21               14  Friday  7.192164e+09          1   
22               14  Friday  7.192148e+09          1   
23               14  Friday  8.830962e+10          1   
24               14  Friday  7.208007e+09          1   
25               16  Friday  4.900005e+09          1   
26               16  Friday  1.775426e+09          1   
27               16  Friday  1.356200e+09          1   
28               16  Friday  6.843791e+09          1   
29               16  Friday  4.900005e+09          1   
...             ...     ...           ...        ...   
653616       191338  Sunday  4.900005e+09          1   
653617       191338  Sunday  4.900005e+09          1   
653618       191338  Sunday  2.420009e+09          6   
653619       191338  Sunday  2.420009e+09          1   
653620       191338  Sunday  2.420009e+09          1   
653621       191338  Sunday  1.070013e+09          1   
653622       191338  Sunday  3.500045e+09          1   
653623       191339  Sunday  4.142011e+09          1   
653624       191339  Sunday  7.245041e+10          1   
653625       191339  Sunday  7.535305e+09          1   
653626       191339  Sunday  7.378578e+10          1   
653627       191339  Sunday  1.600049e+09          2   
653628       191340  Sunday  9.800123e+08          2   
653629       191341  Sunday  3.500039e+09          1   
653630       191341  Sunday  7.935738e+10          2   
653631       191341  Sunday  1.920077e+09          1   
653632       191341  Sunday  7.874211e+09          1   
653633       191341  Sunday  3.700051e+09          1   
653634       191341  Sunday  7.146310e+09          1   
653635       191341  Sunday  3.600044e+09          1   
653636       191341  Sunday  8.861669e+10          1   
653637       191341  Sunday  7.352824e+10          2   
653638       191341  Sunday  8.861669e+10          1   
653639       191348  Sunday  8.046920e+10          1   
653640       191348  Sunday  8.874396e+10          1   
653641       191348  Sunday  6.657211e+10          1   
653642       191348  Sunday  8.818139e+10          1   
653643       191348  Sunday  4.282557e+09          1   
653644       191348  Sunday  8.046919e+10          1   
653645       191348  Sunday  7.871536e+09          1   

              DepartmentDescription  FinelineNumber  
0                             SHOES          3002.0  
1                             DAIRY          1526.0  
2                             DAIRY          1431.0  
3                 GROCERY DRY GOODS          3555.0  
4                       DSD GROCERY          4408.0  
5                            BAKERY          5019.0  
6               IMPULSE MERCHANDISE          8023.0  
7               IMPULSE MERCHANDISE          8023.0  
8                     PERSONAL CARE          4504

In [3]:
train = pd.read_csv('C:\\dss\\team_project2\\walmart\\train.csv')
train

TripType  VisitNumber Weekday           Upc  ScanCount  \
0            999            5  Friday  6.811315e+10         -1   
1             30            7  Friday  6.053882e+10          1   
2             30            7  Friday  7.410811e+09          1   
3             26            8  Friday  2.238404e+09          2   
4             26            8  Friday  2.006614e+09          2   
5             26            8  Friday  2.006619e+09          2   
6             26            8  Friday  2.006614e+09          1   
7             26            8  Friday  7.004803e+09          1   
8             26            8  Friday  2.238495e+09          1   
9             26            8  Friday  2.238400e+09         -1   
10            26            8  Friday  5.200010e+09          1   
11            26            8  Friday  8.867930e+10          2   
12            26            8  Friday  2.200600e+10          1   
13            26            8  Friday  2.236760e+09          1   
14            26            8  Friday  8.867930e+10         -1   
15            26            8  Friday  2.238400e+09          2   
16            26            8  Friday  3.019294e+09          1   
17            26            8  Friday  7.245041e+10          1   
18            26            8  Friday  2.554150e+10          2   
19            26            8  Friday  2.310011e+09          1   
20            26            8  Friday  7.245040e+10          2   
21            26            8  Friday  7.874205e+09          1   
22            26            8  Friday  5.114139e+09          1   
23            26            8  Friday  5.114198e+09          1   
24            26            8  Friday  3.270011e+09          3   
25            26            8  Friday           NaN          1   
26             8            9  Friday  1.070081e+09          1   
27             8            9  Friday  3.107000e+03          1   
28             8            9  Friday  4.011000e+03          1   
29             8           10  Friday  6.414410e+09          1   
...          ...          ...     ...           ...        ...   
647024        39       191345  Sunday  3.800405e+10          2   
647025        39       191345  Sunday  3.890003e+09          1   
647026        39       191345  Sunday  3.890003e+09          1   
647027        39       191345  Sunday  7.225004e+09          1   
647028        39       191345  Sunday  7.800001e+09          1   
647029        39       191345  Sunday  1.380310e+09          1   
647030        39       191345  Sunday  4.812129e+09          2   
647031        39       191345  Sunday  3.029939e+10          1   
647032        39       191345  Sunday  3.010078e+09          2   
647033        39       191345  Sunday  7.286800e+09          1   
647034        39       191345  Sunday  3.700088e+09          1   
647035        39       191346  Sunday  3.760012e+09          1   
647036        39       191346  Sunday  3.338365e+09          1   
647037        39       191346  Sunday  4.150000e+09          1   
647038        39       191346  Sunday  1.340952e+09          1   
647039        39       191346  Sunday  7.874223e+09          1   
647040        39       191346  Sunday  3.700046e+09          1   
647041        39       191346  Sunday  5.440000e+09          1   
647042        39       191346  Sunday  2.265572e+09          1   
647043        39       191346  Sunday  3.660230e+09          1   
647044        39       191346  Sunday  5.100020e+09          1   
647045        39       191346  Sunday  7.874204e+09          1   
647046        39       191346  Sunday  3.120020e+09          1   
647047        39       191346  Sunday  3.120033e+09          1   
647048        39       191346  Sunday  3.700091e+09          1   
647049        39       191346  Sunday  3.239000e+10          1   
647050        39       191346  Sunday  7.874205e+09          1   
647051        39       191346  Sunday  4.072000e+03          1   
647052         8       191347  Sunday  4.190008e

In [4]:
train.isna().sum()

TripType                    0
VisitNumber                 0
Weekday                     0
Upc                      4129
ScanCount                   0
DepartmentDescription    1361
FinelineNumber           4129
dtype: int64

In [5]:
df = train.copy()
del df["Upc"]

In [6]:
df['TripType'].value_counts().sort_index()

3        6827
4         901
5       13836
6        3405
7       23199
8       22844
9       16820
12       2108
14         35
15       7147
18       2977
19       1188
20       3116
21       4032
22       3592
23        325
24      18015
25      27609
26       2507
27       4613
28       2664
29       2105
30       4861
31       1765
32      13843
33       9918
34       4751
35      12501
36      21990
37      38954
38      29565
39      95504
40     174164
41       5508
42      19468
43       6383
44      20424
999     17590
Name: TripType, dtype: int64

### TripType == 31
- WIRELESS
- Saturday가 이기긴했는데....
- 9998

In [7]:
df[df["TripType"] == 31].DepartmentDescription.value_counts()

WIRELESS                       989
IMPULSE MERCHANDISE            121
DSD GROCERY                     70
PERSONAL CARE                   52
GROCERY DRY GOODS               51
PHARMACY OTC                    42
PRODUCE                         38
HOUSEHOLD CHEMICALS/SUPP        30
DAIRY                           29
FINANCIAL SERVICES              27
CANDY, TOBACCO, COOKIES         20
ELECTRONICS                     19
BEAUTY                          19
MENS WEAR                       19
TOYS                            18
PETS AND SUPPLIES               18
LIQUOR,WINE,BEER                17
HOUSEHOLD PAPER GOODS           17
FROZEN FOODS                    14
MEDIA AND GAMING                12
SERVICE DELI                    10
OFFICE SUPPLIES                 10
COMM BREAD                       9
AUTOMOTIVE                       9
CELEBRATION                      9
PLAYERS AND ELECTRONICS          9
SPORTING GOODS                   7
HOME MANAGEMENT                  7
BAKERY              

In [31]:
(df[df["TripType"] == 31].DepartmentDescription.value_counts()/df.DepartmentDescription.value_counts()).sort_values(ascending=False)

WIRELESS                       0.521075
SEASONAL                       0.034483
PLAYERS AND ELECTRONICS        0.013120
MEDIA AND GAMING               0.006021
ELECTRONICS                    0.005956
CAMERAS AND SUPPLIES           0.004525
LIQUOR,WINE,BEER               0.004403
IMPULSE MERCHANDISE            0.004214
ACCESSORIES                    0.003035
MENSWEAR                       0.002959
TOYS                           0.002947
LADIES SOCKS                   0.002581
1-HR PHOTO                     0.002538
FINANCIAL SERVICES             0.002518
CANDY, TOBACCO, COOKIES        0.002039
PETS AND SUPPLIES              0.001844
PHARMACY OTC                   0.001802
PAINT AND ACCESSORIES          0.001744
AUTOMOTIVE                     0.001662
MENS WEAR                      0.001551
OFFICE SUPPLIES                0.001520
SPORTING GOODS                 0.001376
BEAUTY                         0.001278
PERSONAL CARE                  0.001239
HOUSEHOLD CHEMICALS/SUPP       0.001206


In [10]:
df[df["TripType"] == 31].Weekday.value_counts()

Saturday     326
Friday       303
Sunday       303
Thursday     237
Wednesday    226
Monday       212
Tuesday      158
Name: Weekday, dtype: int64

In [27]:
(df[df["TripType"] == 31].Weekday.value_counts()/df.Weekday.value_counts()).sort_values(ascending=False)

Thursday     0.003487
Wednesday    0.003178
Friday       0.003148
Saturday     0.002670
Monday       0.002550
Sunday       0.002262
Tuesday      0.002178
Name: Weekday, dtype: float64

In [33]:
df[df["TripType"] == 31].FinelineNumber.value_counts()

9998.0    302
1165.0    130
965.0      57
1748.0     45
970.0      39
265.0      32
865.0      27
790.0      26
765.0      25
0.0        24
1190.0     23
165.0      20
190.0      19
1265.0     17
990.0      17
1290.0     17
290.0      15
170.0      13
1065.0     13
770.0      11
565.0      10
940.0      10
5501.0      9
880.0       9
1755.0      9
669.0       8
1436.0      8
750.0       8
2007.0      8
490.0       7
         ... 
1040.0      1
103.0       1
101.0       1
1552.0      1
2400.0      1
1329.0      1
2070.0      1
716.0       1
2403.0      1
903.0       1
1311.0      1
9680.0      1
3052.0      1
5105.0      1
2101.0      1
7801.0      1
5230.0      1
2045.0      1
1741.0      1
6405.0      1
1099.0      1
1121.0      1
8956.0      1
2154.0      1
9130.0      1
9625.0      1
4942.0      1
2301.0      1
8007.0      1
5562.0      1
Name: FinelineNumber, Length: 529, dtype: int64

In [37]:
(df[df["TripType"] == 31].FinelineNumber.value_counts()/df.FinelineNumber.value_counts()).sort_values(ascending=False)

1890.0    1.000000
1765.0    1.000000
1140.0    1.000000
490.0     1.000000
1865.0    1.000000
45.0      1.000000
1877.0    1.000000
880.0     0.900000
1190.0    0.884615
990.0     0.850000
950.0     0.800000
790.0     0.787879
940.0     0.769231
465.0     0.750000
9998.0    0.734793
965.0     0.721519
1566.0    0.666667
290.0     0.555556
1165.0    0.550847
1748.0    0.511364
1050.0    0.500000
662.0     0.500000
1090.0    0.466667
865.0     0.435484
970.0     0.433333
870.0     0.428571
890.0     0.333333
1876.0    0.333333
637.0     0.333333
1465.0    0.333333
            ...   
9920.0         NaN
9921.0         NaN
9922.0         NaN
9923.0         NaN
9924.0         NaN
9925.0         NaN
9927.0         NaN
9929.0         NaN
9930.0         NaN
9932.0         NaN
9933.0         NaN
9934.0         NaN
9936.0         NaN
9937.0         NaN
9938.0         NaN
9941.0         NaN
9942.0         NaN
9943.0         NaN
9944.0         NaN
9960.0         NaN
9961.0         NaN
9964.0      

In [85]:
df[df["TripType"] == 31].ScanCount.value_counts()

 1     1574
 2      131
-1       42
 3        8
 4        4
-2        2
 6        2
 10       1
 5        1
Name: ScanCount, dtype: int64

In [40]:
(df[df["TripType"] == 31].ScanCount.value_counts()/df.ScanCount.value_counts()).sort_values(ascending=False).head()

 10    0.007692
-1     0.002918
 1     0.002803
 2     0.002472
-2     0.002457
Name: ScanCount, dtype: float64

### TripType == 32
- INFANT CONSUMABLE HARDLINES, INFANT APPAREL
- 3175

In [52]:
df[df["TripType"] == 32].DepartmentDescription.value_counts()

INFANT CONSUMABLE HARDLINES    4713
INFANT APPAREL                 2415
DSD GROCERY                     879
PERSONAL CARE                   577
IMPULSE MERCHANDISE             572
GROCERY DRY GOODS               475
DAIRY                           434
PRODUCE                         339
HOUSEHOLD CHEMICALS/SUPP        311
SHOES                           265
PHARMACY OTC                    255
CELEBRATION                     199
BEAUTY                          190
TOYS                            186
HOUSEHOLD PAPER GOODS           163
COMM BREAD                      138
FROZEN FOODS                    111
GIRLS WEAR, 4-6X  AND 7-14      103
LADIESWEAR                       95
SERVICE DELI                     92
FINANCIAL SERVICES               91
CANDY, TOBACCO, COOKIES          88
BAKERY                           79
OFFICE SUPPLIES                  76
MENS WEAR                        68
FABRICS AND CRAFTS               66
BOYS WEAR                        59
COOK AND DINE               

In [41]:
(df[df["TripType"] == 32].DepartmentDescription.value_counts()/df.DepartmentDescription.value_counts()).sort_values(ascending=False)

INFANT CONSUMABLE HARDLINES    0.442701
INFANT APPAREL                 0.399108
SHOES                          0.043450
SEASONAL                       0.034483
TOYS                           0.030457
LADIES SOCKS                   0.028387
CELEBRATION                    0.022879
GIRLS WEAR, 4-6X  AND 7-14     0.022808
MENSWEAR                       0.020710
BRAS & SHAPEWEAR               0.020138
IMPULSE MERCHANDISE            0.019922
MEDIA AND GAMING               0.018565
JEWELRY AND SUNGLASSES         0.015439
BOYS WEAR                      0.015159
SLEEPWEAR/FOUNDATIONS          0.014310
PERSONAL CARE                  0.013748
ACCESSORIES                    0.013657
FABRICS AND CRAFTS             0.013067
DSD GROCERY                    0.012864
BEAUTY                         0.012782
HOUSEHOLD CHEMICALS/SUPP       0.012500
HOME DECOR                     0.012479
BOOKS AND MAGAZINES            0.012097
OFFICE SUPPLIES                0.011554
BAKERY                         0.011021


In [53]:
df[df["TripType"] == 32].Weekday.value_counts()

Sunday       2774
Saturday     2459
Friday       2215
Tuesday      1782
Monday       1646
Wednesday    1639
Thursday     1328
Name: Weekday, dtype: int64

In [42]:
(df[df["TripType"] == 32].Weekday.value_counts()/df.Weekday.value_counts()).sort_values(ascending=False)

Tuesday      0.024569
Wednesday    0.023047
Friday       0.023014
Sunday       0.020705
Saturday     0.020140
Monday       0.019800
Thursday     0.019540
Name: Weekday, dtype: float64

In [83]:
df[df["TripType"] == 32].FinelineNumber.value_counts()

3175.0    377
723.0     261
3155.0    260
225.0     148
6013.0    136
3168.0    130
203.0     119
3156.0    118
3163.0    114
1009.0    114
3176.0    113
202.0     112
1001.0    104
6106.0    100
272.0      92
808.0      90
135.0      89
705.0      88
0.0        87
5501.0     86
1508.0     85
1400.0     84
3167.0     79
403.0      73
9546.0     72
2300.0     72
596.0      68
2315.0     67
3192.0     63
804.0      61
         ... 
2650.0      1
6390.0      1
5806.0      1
6102.0      1
2270.0      1
1820.0      1
8076.0      1
8238.0      1
5457.0      1
5504.0      1
90.0        1
5757.0      1
8265.0      1
4902.0      1
8228.0      1
8026.0      1
820.0       1
6365.0      1
2514.0      1
5662.0      1
6312.0      1
8284.0      1
3452.0      1
8300.0      1
4156.0      1
8266.0      1
8040.0      1
559.0       1
3617.0      1
7129.0      1
Name: FinelineNumber, Length: 1947, dtype: int64

In [86]:
df[df["TripType"] == 32].ScanCount.value_counts()

 1     12003
 2      1091
 3       259
 4       203
-1       148
 5        73
 6        18
 7        11
-2         8
 10        7
 8         7
-3         3
 9         3
 11        2
 12        2
-4         1
 18        1
 13        1
 22        1
 71        1
Name: ScanCount, dtype: int64

### TripType == 33
- HOUSEHOKD CHEMICALS/SUPP, HOUSEHOLD PAPER GOODS (가정용 화학용품)
- 8945 (거이 의미가 없는데..)

In [54]:
df[df["TripType"] == 33].DepartmentDescription.value_counts()

HOUSEHOLD CHEMICALS/SUPP       4067
HOUSEHOLD PAPER GOODS          1428
DSD GROCERY                     703
PERSONAL CARE                   559
GROCERY DRY GOODS               463
IMPULSE MERCHANDISE             353
PRODUCE                         333
DAIRY                           250
PHARMACY OTC                    147
BEAUTY                          122
HOME MANAGEMENT                 117
CANDY, TOBACCO, COOKIES         110
FROZEN FOODS                     95
COMM BREAD                       89
BATH AND SHOWER                  87
HARDWARE                         79
SERVICE DELI                     61
COOK AND DINE                    60
INFANT CONSUMABLE HARDLINES      54
OFFICE SUPPLIES                  53
BAKERY                           52
FINANCIAL SERVICES               48
PETS AND SUPPLIES                43
PRE PACKED DELI                  42
HOME DECOR                       38
CELEBRATION                      38
MENS WEAR                        37
MEAT - FRESH & FROZEN       

In [6]:
(df[df["TripType"] == 33].DepartmentDescription.value_counts()/df.DepartmentDescription.value_counts()).sort_values(ascending=False)

HOUSEHOLD CHEMICALS/SUPP       0.163465
HOUSEHOLD PAPER GOODS          0.087747
BATH AND SHOWER                0.018987
HOME MANAGEMENT                0.018428
HARDWARE                       0.016080
PERSONAL CARE                  0.013319
IMPULSE MERCHANDISE            0.012295
CANDY, TOBACCO, COOKIES        0.011215
DSD GROCERY                    0.010288
SHEER HOSIERY                  0.009217
HOME DECOR                     0.009119
BOOKS AND MAGAZINES            0.009073
MEDIA AND GAMING               0.008530
BEAUTY                         0.008207
SWIMWEAR/OUTERWEAR             0.008140
OFFICE SUPPLIES                0.008057
COOK AND DINE                  0.007895
LADIES SOCKS                   0.007742
PLAYERS AND ELECTRONICS        0.007289
BAKERY                         0.007254
GROCERY DRY GOODS              0.006577
PRODUCE                        0.006515
PHARMACY OTC                   0.006307
SERVICE DELI                   0.006024
COMM BREAD                     0.005838


In [56]:
df[df["TripType"] == 33].Weekday.value_counts()

Sunday       1885
Saturday     1687
Friday       1602
Wednesday    1316
Monday       1305
Tuesday      1128
Thursday      995
Name: Weekday, dtype: int64

In [7]:
(df[df["TripType"] == 33].Weekday.value_counts()/df.Weekday.value_counts()).sort_values(ascending=False)

Wednesday    0.018505
Friday       0.016645
Monday       0.015698
Tuesday      0.015552
Thursday     0.014641
Sunday       0.014070
Saturday     0.013817
Name: Weekday, dtype: float64

In [84]:
df[df["TripType"] == 33].FinelineNumber.value_counts()

8945.0    183
55.0      174
8910.0    154
110.0     144
130.0     115
3011.0    113
1006.0    108
5510.0    108
3004.0    102
8599.0     95
2070.0     95
1555.0     91
6010.0     90
1025.0     87
6510.0     79
4510.0     78
3610.0     75
145.0      75
2041.0     70
5501.0     70
4010.0     70
140.0      69
585.0      66
705.0      64
3006.0     64
135.0      63
60.0       60
808.0      60
1014.0     60
235.0      60
         ... 
1904.0      1
2024.0      1
4034.0      1
276.0       1
7805.0      1
6604.0      1
3818.0      1
9545.0      1
2993.0      1
2115.0      1
1107.0      1
4863.0      1
5412.0      1
2457.0      1
1552.0      1
4235.0      1
1813.0      1
75.0        1
3766.0      1
4605.0      1
3252.0      1
7502.0      1
2586.0      1
1111.0      1
9551.0      1
6641.0      1
3149.0      1
7148.0      1
6230.0      1
2705.0      1
Name: FinelineNumber, Length: 1515, dtype: int64

In [44]:
(df[df["TripType"] == 33].FinelineNumber.value_counts()/df.FinelineNumber.value_counts()).sort_values(ascending=False)

9718.0    1.000000
9714.0    1.000000
9056.0    1.000000
4255.0    0.500000
9709.0    0.500000
9401.0    0.500000
9050.0    0.421053
9053.0    0.411765
9410.0    0.400000
1162.0    0.357143
32.0      0.333333
2304.0    0.333333
9540.0    0.285714
7520.0    0.282258
9054.0    0.258621
9061.0    0.252747
9717.0    0.250000
8408.0    0.250000
9503.0    0.250000
8881.0    0.250000
9030.0    0.244681
9102.0    0.243243
9055.0    0.236842
217.0     0.232558
57.0      0.224359
9301.0    0.222222
8599.0    0.215420
9403.0    0.214876
9712.0    0.214286
958.0     0.212766
            ...   
9922.0         NaN
9923.0         NaN
9924.0         NaN
9925.0         NaN
9927.0         NaN
9928.0         NaN
9929.0         NaN
9930.0         NaN
9931.0         NaN
9932.0         NaN
9933.0         NaN
9934.0         NaN
9936.0         NaN
9937.0         NaN
9938.0         NaN
9941.0         NaN
9942.0         NaN
9943.0         NaN
9944.0         NaN
9960.0         NaN
9961.0         NaN
9964.0      

In [88]:
df[df["TripType"] == 33].ScanCount.value_counts()

 1     8659
 2      865
 3      154
-1       91
 4       78
 6       24
 5       21
 8        7
-2        6
 7        4
 9        3
 10       1
-6        1
 12       1
 25       1
-3        1
 11       1
Name: ScanCount, dtype: int64

In [45]:
(df[df["TripType"] == 33].ScanCount.value_counts()/df.ScanCount.value_counts()).sort_values(ascending=False)

 25    0.333333
-6     0.100000
 9     0.034091
 8     0.028455
 6     0.027714
 11    0.022222
 7     0.017544
 4     0.017196
 12    0.016667
 2     0.016320
 3     0.016305
 1     0.015421
 5     0.015217
 10    0.007692
-2     0.007371
-3     0.006993
-1     0.006322
-12         NaN
-10         NaN
-9          NaN
-7          NaN
-5          NaN
-4          NaN
 13         NaN
 14         NaN
 15         NaN
 16         NaN
 17         NaN
 18         NaN
 19         NaN
 20         NaN
 22         NaN
 23         NaN
 24         NaN
 30         NaN
 31         NaN
 46         NaN
 51         NaN
 71         NaN
Name: ScanCount, dtype: float64

### TripType == 34
- PETS AND SUPPLIES
- Saturday 가 압도적
- 1946 (이것도뭐.. 거이 의미x)

In [57]:
df[df["TripType"] == 34].DepartmentDescription.value_counts()

PETS AND SUPPLIES              2657
DSD GROCERY                     228
IMPULSE MERCHANDISE             192
PERSONAL CARE                   185
GROCERY DRY GOODS               161
PRODUCE                         122
PHARMACY OTC                    111
DAIRY                           105
HOUSEHOLD CHEMICALS/SUPP         90
BEAUTY                           70
FROZEN FOODS                     57
HARDWARE                         50
AUTOMOTIVE                       50
COMM BREAD                       49
CANDY, TOBACCO, COOKIES          49
HOUSEHOLD PAPER GOODS            47
BAKERY                           46
LAWN AND GARDEN                  42
COOK AND DINE                    37
SERVICE DELI                     36
HOME MANAGEMENT                  35
FINANCIAL SERVICES               34
MENS WEAR                        24
HOME DECOR                       23
SHOES                            17
OFFICE SUPPLIES                  17
PRE PACKED DELI                  16
BATH AND SHOWER             

In [46]:
(df[df["TripType"] == 34].DepartmentDescription.value_counts()/df.DepartmentDescription.value_counts()).sort_values(ascending=False)

PETS AND SUPPLIES              0.272206
HARDWARE                       0.010177
AUTOMOTIVE                     0.009235
LAWN AND GARDEN                0.007971
IMPULSE MERCHANDISE            0.006687
BAKERY                         0.006417
HOME DECOR                     0.005520
HOME MANAGEMENT                0.005513
HORTICULTURE AND ACCESS        0.005185
CANDY, TOBACCO, COOKIES        0.004996
COOK AND DINE                  0.004868
PHARMACY OTC                   0.004763
BEAUTY                         0.004709
PERSONAL CARE                  0.004408
JEWELRY AND SUNGLASSES         0.004355
LIQUOR,WINE,BEER               0.003885
HOUSEHOLD CHEMICALS/SUPP       0.003617
SERVICE DELI                   0.003555
DSD GROCERY                    0.003337
BATH AND SHOWER                0.003274
COMM BREAD                     0.003214
FINANCIAL SERVICES             0.003171
MENSWEAR                       0.002959
PAINT AND ACCESSORIES          0.002907
HOUSEHOLD PAPER GOODS          0.002888


In [58]:
df[df["TripType"] == 34].Weekday.value_counts()

Saturday     1020
Friday        788
Sunday        760
Tuesday       646
Monday        585
Thursday      493
Wednesday     459
Name: Weekday, dtype: int64

In [47]:
(df[df["TripType"] == 34].Weekday.value_counts()/df.Weekday.value_counts()).sort_values(ascending=False)

Tuesday      0.008907
Saturday     0.008354
Friday       0.008187
Thursday     0.007254
Monday       0.007037
Wednesday    0.006454
Sunday       0.005673
Name: Weekday, dtype: float64

In [89]:
df[df["TripType"] == 34].FinelineNumber.value_counts()

1946.0    228
1436.0    216
7905.0    124
630.0     104
1835.0     77
1001.0     63
3510.0     59
610.0      57
609.0      46
5501.0     43
5260.0     39
3300.0     37
230.0      37
0.0        34
1020.0     33
7420.0     32
203.0      29
135.0      29
7440.0     29
4801.0     29
9546.0     28
3105.0     28
806.0      27
7612.0     26
240.0      26
7410.0     26
2025.0     25
242.0      25
5830.0     25
5620.0     25
         ... 
84.0        1
436.0       1
8960.0      1
116.0       1
345.0       1
94.0        1
6643.0      1
2601.0      1
2906.0      1
785.0       1
5007.0      1
2019.0      1
5076.0      1
2587.0      1
3870.0      1
8220.0      1
1605.0      1
7106.0      1
4636.0      1
3201.0      1
3810.0      1
2552.0      1
1358.0      1
733.0       1
4261.0      1
5070.0      1
1111.0      1
5555.0      1
7902.0      1
603.0       1
Name: FinelineNumber, Length: 1133, dtype: int64

In [48]:
(df[df["TripType"] == 34].FinelineNumber.value_counts()/df.FinelineNumber.value_counts()).sort_values(ascending=False)

4411.0    0.650000
4211.0    0.545455
7440.0    0.537037
7225.0    0.520000
7210.0    0.500000
3412.0    0.500000
7450.0    0.500000
7420.0    0.500000
7410.0    0.464286
1825.0    0.400000
9852.0    0.400000
2632.0    0.400000
3316.0    0.400000
1435.0    0.394737
7407.0    0.393939
3641.0    0.388889
242.0     0.384615
7905.0    0.382716
4225.0    0.363636
1425.0    0.362069
4410.0    0.352941
1709.0    0.333333
3908.0    0.333333
1876.0    0.333333
5076.0    0.333333
5261.0    0.333333
5636.0    0.326923
845.0     0.321429
3631.0    0.315789
1947.0    0.307692
            ...   
9923.0         NaN
9924.0         NaN
9925.0         NaN
9927.0         NaN
9928.0         NaN
9929.0         NaN
9930.0         NaN
9931.0         NaN
9932.0         NaN
9933.0         NaN
9934.0         NaN
9936.0         NaN
9937.0         NaN
9938.0         NaN
9941.0         NaN
9942.0         NaN
9943.0         NaN
9944.0         NaN
9960.0         NaN
9961.0         NaN
9964.0         NaN
9966.0      

In [90]:
df[df["TripType"] == 34].ScanCount.value_counts()

 1     3987
 2      474
 3      120
 4       50
-1       37
 5       26
 6       21
-2        8
 8        7
 7        4
 10       3
 12       3
 9        3
 11       2
 24       1
 23       1
-3        1
 13       1
 15       1
 25       1
Name: ScanCount, dtype: int64

In [49]:
(df[df["TripType"] == 34].ScanCount.value_counts()/df.ScanCount.value_counts()).sort_values(ascending=False)

 25    0.333333
 23    0.333333
 24    0.250000
 15    0.083333
 13    0.076923
 12    0.050000
 11    0.044444
 9     0.034091
 8     0.028455
 6     0.024249
 10    0.023077
 5     0.018841
 7     0.017544
 3     0.012705
 4     0.011023
-2     0.009828
 2     0.008943
 1     0.007101
-3     0.006993
-1     0.002571
-12         NaN
-10         NaN
-9          NaN
-7          NaN
-6          NaN
-5          NaN
-4          NaN
 14         NaN
 16         NaN
 17         NaN
 18         NaN
 19         NaN
 20         NaN
 22         NaN
 30         NaN
 31         NaN
 46         NaN
 51         NaN
 71         NaN
Name: ScanCount, dtype: float64

### TripType == 35
- DSD GROCERY
- 808 (too)

In [59]:
df[df["TripType"] == 35].DepartmentDescription.value_counts()

DSD GROCERY                    6135
IMPULSE MERCHANDISE            1757
PRODUCE                         706
GROCERY DRY GOODS               682
CANDY, TOBACCO, COOKIES         437
DAIRY                           406
COMM BREAD                      265
HOUSEHOLD CHEMICALS/SUPP        214
PERSONAL CARE                   207
FROZEN FOODS                    196
BAKERY                          186
HOUSEHOLD PAPER GOODS           186
PHARMACY OTC                    141
SERVICE DELI                    138
PRE PACKED DELI                  77
FINANCIAL SERVICES               68
OFFICE SUPPLIES                  59
INFANT CONSUMABLE HARDLINES      52
CELEBRATION                      45
BEAUTY                           41
MENS WEAR                        39
PETS AND SUPPLIES                36
COOK AND DINE                    36
LIQUOR,WINE,BEER                 31
AUTOMOTIVE                       27
FABRICS AND CRAFTS               27
TOYS                             26
MEAT - FRESH & FROZEN       

In [50]:
(df[df["TripType"] == 35].DepartmentDescription.value_counts()/df.DepartmentDescription.value_counts()).sort_values(ascending=False)

DSD GROCERY                    0.089782
IMPULSE MERCHANDISE            0.061194
CANDY, TOBACCO, COOKIES        0.044555
BAKERY                         0.025949
COMM BREAD                     0.017384
PRODUCE                        0.013812
SERVICE DELI                   0.013627
HOUSEHOLD PAPER GOODS          0.011429
GROCERY DRY GOODS              0.009687
FROZEN FOODS                   0.009289
DAIRY                          0.009265
BOOKS AND MAGAZINES            0.009073
OFFICE SUPPLIES                0.008969
HOUSEHOLD CHEMICALS/SUPP       0.008601
PRE PACKED DELI                0.008506
LIQUOR,WINE,BEER               0.008029
1-HR PHOTO                     0.007614
FINANCIAL SERVICES             0.006342
PHARMACY OTC                   0.006050
FABRICS AND CRAFTS             0.005345
CELEBRATION                    0.005174
LADIES SOCKS                   0.005161
AUTOMOTIVE                     0.004987
PERSONAL CARE                  0.004932
INFANT CONSUMABLE HARDLINES    0.004884


In [60]:
df[df["TripType"] == 35].Weekday.value_counts()

Sunday       2130
Saturday     2084
Friday       1943
Monday       1686
Thursday     1654
Wednesday    1565
Tuesday      1439
Name: Weekday, dtype: int64

In [10]:
(df[df["TripType"] == 35].Weekday.value_counts()/df.Weekday.value_counts()).sort_values(ascending=False)

Thursday     0.024337
Wednesday    0.022007
Monday       0.020281
Friday       0.020188
Tuesday      0.019840
Saturday     0.017069
Sunday       0.015898
Name: Weekday, dtype: float64

In [91]:
df[df["TripType"] == 35].FinelineNumber.value_counts()

808.0     414
4606.0    367
4607.0    274
135.0     256
9546.0    220
115.0     215
100.0     201
4624.0    197
5501.0    164
9205.0    163
3658.0    161
9570.0    128
9516.0    124
9538.0    124
3467.0    124
3004.0    121
804.0     120
9550.0    111
2403.0    108
4611.0    108
3353.0    102
9214.0     95
9515.0     92
3301.0     89
2351.0     84
601.0      81
145.0      80
3054.0     74
110.0      73
3466.0     70
         ... 
718.0       1
5111.0      1
1553.0      1
5502.0      1
8018.0      1
1551.0      1
4602.0      1
5520.0      1
3738.0      1
4230.0      1
3501.0      1
2614.0      1
5970.0      1
2017.0      1
3100.0      1
395.0       1
8138.0      1
4382.0      1
1850.0      1
1531.0      1
8235.0      1
5315.0      1
4405.0      1
2410.0      1
3749.0      1
1135.0      1
2753.0      1
535.0       1
2838.0      1
4159.0      1
Name: FinelineNumber, Length: 1474, dtype: int64

In [11]:
(df[df["TripType"] == 35].FinelineNumber.value_counts()/df.FinelineNumber.value_counts()).sort_values(ascending=False)

3234.0    1.000000
8402.0    0.333333
7555.0    0.333333
9290.0    0.333333
4540.0    0.289474
7317.0    0.285714
4542.0    0.260870
9513.0    0.258427
3567.0    0.250000
1686.0    0.250000
4692.0    0.250000
3629.0    0.250000
4543.0    0.250000
4589.0    0.250000
376.0     0.222222
4552.0    0.222222
4677.0    0.217391
9550.0    0.213462
9538.0    0.207358
2373.0    0.200000
2320.0    0.200000
3233.0    0.195122
9509.0    0.193182
9263.0    0.190476
4607.0    0.187158
9581.0    0.184466
9202.0    0.184211
9551.0    0.184000
4614.0    0.181818
4333.0    0.175000
            ...   
9922.0         NaN
9923.0         NaN
9924.0         NaN
9925.0         NaN
9927.0         NaN
9928.0         NaN
9929.0         NaN
9930.0         NaN
9931.0         NaN
9932.0         NaN
9933.0         NaN
9936.0         NaN
9937.0         NaN
9938.0         NaN
9941.0         NaN
9942.0         NaN
9943.0         NaN
9944.0         NaN
9960.0         NaN
9961.0         NaN
9964.0         NaN
9966.0      

In [92]:
df[df["TripType"] == 35].ScanCount.value_counts()

 1     10122
 2      1566
 3       374
 4       210
-1        95
 5        57
 6        32
 8        11
-2         9
 7         6
 9         5
 12        3
 11        3
 10        3
-4         1
 19        1
-3         1
 14        1
-9         1
Name: ScanCount, dtype: int64

In [12]:
(df[df["TripType"] == 35].ScanCount.value_counts()/df.ScanCount.value_counts()).sort_values(ascending=False)

-9     0.500000
 19    0.250000
 14    0.066667
 11    0.066667
 9     0.056818
 12    0.050000
 4     0.046296
 8     0.044715
 5     0.041304
 3     0.039598
 6     0.036952
 2     0.029546
 7     0.026316
 10    0.023077
 1     0.018027
-4     0.013158
-2     0.011057
-3     0.006993
-1     0.006600
-12         NaN
-10         NaN
-7          NaN
-6          NaN
-5          NaN
 13         NaN
 15         NaN
 16         NaN
 17         NaN
 18         NaN
 20         NaN
 22         NaN
 23         NaN
 24         NaN
 25         NaN
 30         NaN
 31         NaN
 46         NaN
 51         NaN
 71         NaN
Name: ScanCount, dtype: float64

### TripType == 36
- PERSONAL CARE (개인 용품정도)
- Saturday가 Sunday 를 이김
- 203 (애매..)

In [61]:
df[df["TripType"] == 36].DepartmentDescription.value_counts()

PERSONAL CARE                  8635
BEAUTY                         3843
PHARMACY OTC                   1594
IMPULSE MERCHANDISE            1176
DSD GROCERY                    1003
GROCERY DRY GOODS               637
PRODUCE                         541
HOUSEHOLD CHEMICALS/SUPP        464
DAIRY                           347
HOUSEHOLD PAPER GOODS           289
CANDY, TOBACCO, COOKIES         240
CELEBRATION                     201
FROZEN FOODS                    182
PETS AND SUPPLIES               156
MENS WEAR                       149
OFFICE SUPPLIES                 149
HOME MANAGEMENT                 131
FINANCIAL SERVICES              124
BATH AND SHOWER                 120
COMM BREAD                      117
COOK AND DINE                   116
INFANT CONSUMABLE HARDLINES     114
LADIESWEAR                      107
SERVICE DELI                     95
TOYS                             88
AUTOMOTIVE                       88
BAKERY                           83
SHOES                       

In [13]:
(df[df["TripType"] == 36].DepartmentDescription.value_counts()/df.DepartmentDescription.value_counts()).sort_values(ascending=False)

HEALTH AND BEAUTY AIDS         0.500000
BEAUTY                         0.258527
PERSONAL CARE                  0.205747
PHARMACY OTC                   0.068394
IMPULSE MERCHANDISE            0.040958
OPTICAL - FRAMES               0.039724
LADIES SOCKS                   0.030968
SHEER HOSIERY                  0.027650
BATH AND SHOWER                0.026189
JEWELRY AND SUNGLASSES         0.025337
CANDY, TOBACCO, COOKIES        0.024470
ACCESSORIES                    0.023520
CELEBRATION                    0.023109
OFFICE SUPPLIES                0.022651
SLEEPWEAR/FOUNDATIONS          0.021465
BOOKS AND MAGAZINES            0.021169
HOME MANAGEMENT                0.020633
HORTICULTURE AND ACCESS        0.018666
HOUSEHOLD CHEMICALS/SUPP       0.018650
HOME DECOR                     0.017999
1-HR PHOTO                     0.017766
HOUSEHOLD PAPER GOODS          0.017758
AUTOMOTIVE                     0.016254
PETS AND SUPPLIES              0.015982
FABRICS AND CRAFTS             0.015442


In [62]:
df[df["TripType"] == 36].Weekday.value_counts()

Saturday     4139
Sunday       3807
Friday       3583
Monday       2793
Thursday     2735
Wednesday    2512
Tuesday      2421
Name: Weekday, dtype: int64

In [93]:
df[df["TripType"] == 36].FinelineNumber.value_counts()

203.0     415
4010.0    286
135.0     270
403.0     226
8101.0    221
801.0     212
4020.0    207
9108.0    194
6124.0    189
202.0     189
3181.0    175
5050.0    172
5501.0    166
808.0     156
1403.0    153
6080.0    139
1411.0    138
4501.0    135
53.0      134
3100.0    132
2105.0    124
0.0       120
205.0     116
402.0     116
4503.0    114
2230.0    112
4500.0    111
4510.0    104
4541.0    101
9546.0    101
         ... 
5410.0      1
2013.0      1
4301.0      1
9268.0      1
2532.0      1
8167.0      1
5433.0      1
8162.0      1
4863.0      1
422.0       1
1066.0      1
3758.0      1
7220.0      1
2215.0      1
6097.0      1
2832.0      1
146.0       1
4374.0      1
8403.0      1
5903.0      1
4033.0      1
7810.0      1
7916.0      1
7022.0      1
3252.0      1
8023.0      1
2808.0      1
8248.0      1
4120.0      1
2155.0      1
Name: FinelineNumber, Length: 2267, dtype: int64

In [94]:
df[df["TripType"] == 36].ScanCount.value_counts()

 1     19793
 2      1498
 3       261
-1       177
 4       135
 5        53
-2        19
 6        18
 8         7
 10        7
 7         4
-3         4
-4         3
 12        3
 9         2
 11        2
-6         1
-9         1
 20        1
 31        1
Name: ScanCount, dtype: int64

### TripType == 37
- PRODUCE
- 5501 (압도적)

In [63]:
df[df["TripType"] == 37].DepartmentDescription.value_counts()

PRODUCE                        13351
GROCERY DRY GOODS               5477
DAIRY                           3667
DSD GROCERY                     3282
MEAT - FRESH & FROZEN           2596
COMM BREAD                      1150
FROZEN FOODS                    1102
PRE PACKED DELI                  883
IMPULSE MERCHANDISE              807
HOUSEHOLD CHEMICALS/SUPP         751
PERSONAL CARE                    696
HOUSEHOLD PAPER GOODS            629
SERVICE DELI                     580
BAKERY                           544
PHARMACY OTC                     437
SEAFOOD                          276
CANDY, TOBACCO, COOKIES          235
BEAUTY                           198
COOK AND DINE                    197
PETS AND SUPPLIES                181
LIQUOR,WINE,BEER                 165
INFANT CONSUMABLE HARDLINES      153
FINANCIAL SERVICES               122
LAWN AND GARDEN                  115
LADIESWEAR                       100
MENS WEAR                         88
CELEBRATION                       87
T

In [64]:
df[df["TripType"] == 37].Weekday.value_counts()

Sunday       9119
Saturday     6774
Friday       5678
Monday       5405
Tuesday      4247
Thursday     3913
Wednesday    3818
Name: Weekday, dtype: int64

In [95]:
df[df["TripType"] == 37].FinelineNumber.value_counts()

5501.0    1365
3601.0     564
5620.0     476
7803.0     469
8101.0     327
1508.0     325
8104.0     302
5701.0     294
3004.0     268
7501.0     262
6201.0     235
8303.0     229
1407.0     210
4008.0     206
135.0      190
3729.0     185
7955.0     180
2005.0     167
8001.0     165
8400.0     165
1201.0     162
9101.0     160
2500.0     159
1301.0     157
2400.0     154
1515.0     152
7907.0     151
5017.0     151
2351.0     147
3600.0     146
          ... 
4220.0       1
5211.0       1
9009.0       1
3922.0       1
5561.0       1
5039.0       1
1421.0       1
8048.0       1
1222.0       1
2617.0       1
141.0        1
2098.0       1
9290.0       1
4303.0       1
6753.0       1
8029.0       1
1135.0       1
800.0        1
2267.0       1
5117.0       1
9405.0       1
4516.0       1
3618.0       1
8093.0       1
2558.0       1
6751.0       1
4923.0       1
3743.0       1
998.0        1
1439.0       1
Name: FinelineNumber, Length: 2542, dtype: int64

In [96]:
df[df["TripType"] == 37].ScanCount.value_counts()

 1     34782
 2      3066
 3       489
-1       307
 4       171
 5        51
 6        40
-2        23
 7        10
 10        4
 8         3
 12        2
 11        2
 18        1
-4         1
-3         1
 9         1
Name: ScanCount, dtype: int64

### TripType == 38
- DAIRY, GROCERY DRY GOODS
- 월요 의지 불태움
- 1508 (두배정도)

In [65]:
df[df["TripType"] == 38].DepartmentDescription.value_counts()

DAIRY                          7252
GROCERY DRY GOODS              6417
DSD GROCERY                    4020
PRODUCE                        2161
COMM BREAD                     1662
FROZEN FOODS                   1310
IMPULSE MERCHANDISE             839
PRE PACKED DELI                 807
HOUSEHOLD CHEMICALS/SUPP        629
PERSONAL CARE                   587
BAKERY                          518
HOUSEHOLD PAPER GOODS           490
PHARMACY OTC                    344
CANDY, TOBACCO, COOKIES         322
SERVICE DELI                    251
MEAT - FRESH & FROZEN           202
BEAUTY                          179
INFANT CONSUMABLE HARDLINES     170
PETS AND SUPPLIES               128
FINANCIAL SERVICES              108
CELEBRATION                      90
OFFICE SUPPLIES                  85
COOK AND DINE                    69
LIQUOR,WINE,BEER                 62
LADIESWEAR                       61
TOYS                             57
AUTOMOTIVE                       56
FABRICS AND CRAFTS          

In [66]:
df[df["TripType"] == 38].Weekday.value_counts()

Sunday       6254
Saturday     4363
Monday       4322
Friday       4113
Tuesday      3862
Wednesday    3550
Thursday     3101
Name: Weekday, dtype: int64

In [97]:
df[df["TripType"] == 38].FinelineNumber.value_counts()

1508.0    1008
3600.0     590
3555.0     571
5501.0     538
1407.0     441
9100.0     370
7955.0     279
1443.0     278
3724.0     277
7978.0     247
3559.0     245
1404.0     243
1450.0     214
7962.0     208
4644.0     200
3556.0     196
9101.0     194
2112.0     194
1515.0     174
1351.0     169
3120.0     163
7958.0     161
3701.0     159
808.0      157
3707.0     155
3601.0     151
7953.0     140
4624.0     139
4639.0     138
135.0      137
          ... 
3166.0       1
633.0        1
7300.0       1
6809.0       1
193.0        1
3912.0       1
4083.0       1
1651.0       1
2157.0       1
1057.0       1
7815.0       1
933.0        1
7024.0       1
6024.0       1
107.0        1
8117.0       1
5540.0       1
8305.0       1
7609.0       1
225.0        1
3432.0       1
706.0        1
2619.0       1
6044.0       1
41.0         1
6350.0       1
6103.0       1
543.0        1
2518.0       1
5007.0       1
Name: FinelineNumber, Length: 2117, dtype: int64

In [98]:
df[df["TripType"] == 38].ScanCount.value_counts()

 1     25490
 2      2910
 3       495
 4       271
-1       208
 5        90
 6        48
-2        12
 7        10
 8         8
 10        6
 12        4
 9         3
 11        3
-3         2
 24        1
-4         1
 13        1
 46        1
 15        1
Name: ScanCount, dtype: int64

### TripType == 39
- DSD GROCERY, GROCERY DRY GOODS (두개 차이가 거이없음)
- 5501 (거이 두배)

In [67]:
df[df["TripType"] == 39].DepartmentDescription.value_counts()

DSD GROCERY                    12956
GROCERY DRY GOODS              12602
PERSONAL CARE                   8126
DAIRY                           6895
PRODUCE                         6490
HOUSEHOLD CHEMICALS/SUPP        5181
PHARMACY OTC                    4617
FROZEN FOODS                    4187
IMPULSE MERCHANDISE             4100
HOUSEHOLD PAPER GOODS           3498
COMM BREAD                      2648
CANDY, TOBACCO, COOKIES         2327
BEAUTY                          2214
PETS AND SUPPLIES               1598
SERVICE DELI                    1579
PRE PACKED DELI                 1507
BAKERY                          1367
INFANT CONSUMABLE HARDLINES     1177
MEAT - FRESH & FROZEN            959
LIQUOR,WINE,BEER                 883
MENS WEAR                        779
COOK AND DINE                    727
CELEBRATION                      580
HOME MANAGEMENT                  570
SHOES                            501
OFFICE SUPPLIES                  489
LADIESWEAR                       482
L

In [68]:
df[df["TripType"] == 39].Weekday.value_counts()

Sunday       19444
Saturday     16708
Friday       14551
Monday       12884
Tuesday      11005
Thursday     10549
Wednesday    10363
Name: Weekday, dtype: int64

In [99]:
df[df["TripType"] == 39].FinelineNumber.value_counts()

5501.0    1382
1508.0     712
135.0      702
808.0      689
203.0      544
1407.0     534
4606.0     526
3004.0     504
115.0      497
9546.0     454
0.0        436
110.0      431
4010.0     407
100.0      391
5017.0     387
3601.0     358
1851.0     353
8101.0     340
3011.0     338
7010.0     336
4624.0     329
3701.0     326
3702.0     326
202.0      321
3600.0     320
9570.0     319
9205.0     317
1853.0     316
8910.0     305
7955.0     300
          ... 
853.0        1
5072.0       1
1735.0       1
5434.0       1
8922.0       1
4275.0       1
4012.0       1
733.0        1
7225.0       1
4289.0       1
2037.0       1
4086.0       1
1955.0       1
9938.0       1
6477.0       1
4967.0       1
8099.0       1
361.0        1
9507.0       1
3759.0       1
2635.0       1
3266.0       1
9936.0       1
3900.0       1
2928.0       1
7055.0       1
5581.0       1
1071.0       1
679.0        1
2202.0       1
Name: FinelineNumber, Length: 3864, dtype: int64

In [100]:
df[df["TripType"] == 39].ScanCount.value_counts()

 1     85603
 2      7134
 3      1156
-1       719
 4       570
 5       144
 6        87
-2        36
 8        12
 7         9
-3         8
 12        6
 9         6
 10        6
-4         4
-5         1
-12        1
 15        1
 11        1
Name: ScanCount, dtype: int64

### TripType == 40
- GROCERY DRY GOODS, DSD GROCERY (10000개정도 차이나는데 둘다많음)
- 5501

In [69]:
df[df["TripType"] == 40].DepartmentDescription.value_counts()

GROCERY DRY GOODS              32639
DSD GROCERY                    22157
DAIRY                          17354
PRODUCE                        16704
PERSONAL CARE                   9911
FROZEN FOODS                    9307
HOUSEHOLD CHEMICALS/SUPP        8093
COMM BREAD                      5968
HOUSEHOLD PAPER GOODS           5712
PHARMACY OTC                    4569
PRE PACKED DELI                 4134
IMPULSE MERCHANDISE             4106
MEAT - FRESH & FROZEN           3557
PETS AND SUPPLIES               2792
BEAUTY                          2697
CANDY, TOBACCO, COOKIES         2616
INFANT CONSUMABLE HARDLINES     2037
SERVICE DELI                    1980
BAKERY                          1693
COOK AND DINE                   1389
LIQUOR,WINE,BEER                 941
CELEBRATION                      834
SEAFOOD                          822
HOME MANAGEMENT                  773
LADIESWEAR                       765
MENS WEAR                        736
LAWN AND GARDEN                  716
O

In [70]:
df[df["TripType"] == 40].Weekday.value_counts()

Sunday       43024
Saturday     36383
Friday       23002
Monday       22380
Tuesday      17954
Wednesday    16748
Thursday     14673
Name: Weekday, dtype: int64

In [101]:
df[df["TripType"] == 40].FinelineNumber.value_counts()

5501.0    2293
1508.0    1553
1407.0    1036
3600.0    1000
3120.0     996
4606.0     851
3601.0     790
7955.0     779
808.0      756
4624.0     724
9101.0     720
1750.0     719
3004.0     691
2995.0     684
135.0      678
3701.0     666
3555.0     656
7958.0     653
203.0      650
3702.0     631
7554.0     623
5620.0     621
1450.0     620
8101.0     618
2453.0     617
110.0      605
3724.0     599
1436.0     594
1901.0     585
4451.0     565
          ... 
747.0        1
8961.0       1
154.0        1
8008.0       1
5361.0       1
2738.0       1
1249.0       1
3280.0       1
2517.0       1
9125.0       1
9124.0       1
1419.0       1
1936.0       1
6607.0       1
1156.0       1
2131.0       1
2284.0       1
392.0        1
2668.0       1
1072.0       1
7155.0       1
2669.0       1
7056.0       1
5340.0       1
587.0        1
5351.0       1
1428.0       1
333.0        1
3915.0       1
4833.0       1
Name: FinelineNumber, Length: 4062, dtype: int64

In [102]:
df[df["TripType"] == 40].ScanCount.value_counts()

 1     150250
 2      17209
 3       3104
 4       1468
-1       1034
 5        410
 6        302
 8         89
 7         77
-2         66
 10        45
 9         26
 12        19
 11        19
-3         14
 14         7
-4          6
 13         4
 16         4
 15         3
 19         3
 24         2
 20         1
 51         1
 17         1
Name: ScanCount, dtype: int64

### TripType == 41 - onemore...
- SHOES, IMPULSE MERCHANDISE, PERSONAL CARE, DSD GROCERY... (차이가 너무 없음)
- 135 (의미없음)

In [71]:
df[df["TripType"] == 41].DepartmentDescription.value_counts()

SHOES                          340
IMPULSE MERCHANDISE            328
PERSONAL CARE                  326
DSD GROCERY                    300
MENS WEAR                      270
HARDWARE                       209
COOK AND DINE                  199
SPORTING GOODS                 188
INFANT APPAREL                 152
BATH AND SHOWER                147
TOYS                           142
GROCERY DRY GOODS              140
PHARMACY OTC                   139
GIRLS WEAR, 4-6X  AND 7-14     134
LADIESWEAR                     124
HOME DECOR                     123
CELEBRATION                    120
BEAUTY                         118
HOUSEHOLD CHEMICALS/SUPP       118
BOYS WEAR                      113
HOME MANAGEMENT                112
AUTOMOTIVE                     105
LAWN AND GARDEN                 99
INFANT CONSUMABLE HARDLINES     96
OFFICE SUPPLIES                 95
JEWELRY AND SUNGLASSES          94
ELECTRONICS                     93
DAIRY                           82
ACCESSORIES         

In [72]:
df[df["TripType"] == 41].Weekday.value_counts()

Sunday       1285
Saturday     1218
Friday        862
Thursday      659
Monday        562
Wednesday     501
Tuesday       421
Name: Weekday, dtype: int64

In [103]:
df[df["TripType"] == 41].FinelineNumber.value_counts()

135.0     48
702.0     41
808.0     39
0.0       33
100.0     32
302.0     28
9546.0    24
3467.0    24
5501.0    23
145.0     23
115.0     22
804.0     21
202.0     21
1202.0    20
1211.0    18
801.0     18
8960.0    17
1805.0    17
4520.0    16
1505.0    15
203.0     15
211.0     15
250.0     15
581.0     15
4606.0    14
9211.0    14
5005.0    14
6080.0    14
8001.0    14
1555.0    13
          ..
2428.0     1
6670.0     1
3730.0     1
5700.0     1
4106.0     1
1213.0     1
1850.0     1
7055.0     1
3099.0     1
8933.0     1
6191.0     1
2456.0     1
4170.0     1
3231.0     1
8133.0     1
3599.0     1
5138.0     1
6408.0     1
2991.0     1
6476.0     1
1286.0     1
4307.0     1
3324.0     1
7713.0     1
3853.0     1
6552.0     1
1162.0     1
2303.0     1
3462.0     1
3185.0     1
Name: FinelineNumber, Length: 1969, dtype: int64

In [104]:
df[df["TripType"] == 41].ScanCount.value_counts()

 1     4968
 2      370
 3       60
-1       59
 4       29
 5       10
 6        5
 7        2
-2        2
 10       1
 8        1
-4        1
Name: ScanCount, dtype: int64

### TripType == 42
- IMPULSE MERCHANDISE, CELEBRATION (차이가 크지않음)
- 1805 (의미x)

In [73]:
df[df["TripType"] == 42].DepartmentDescription.value_counts()

IMPULSE MERCHANDISE            1278
CELEBRATION                    1137
PERSONAL CARE                   860
MENS WEAR                       857
INFANT APPAREL                  849
TOYS                            845
DSD GROCERY                     830
OFFICE SUPPLIES                 729
LADIESWEAR                      679
FABRICS AND CRAFTS              654
SHOES                           619
GROCERY DRY GOODS               484
GIRLS WEAR, 4-6X  AND 7-14      481
LAWN AND GARDEN                 447
BEAUTY                          444
BATH AND SHOWER                 440
HOME MANAGEMENT                 440
PHARMACY OTC                    436
COOK AND DINE                   434
SPORTING GOODS                  428
BOYS WEAR                       408
HARDWARE                        385
AUTOMOTIVE                      376
HOUSEHOLD CHEMICALS/SUPP        344
PRODUCE                         335
HOME DECOR                      311
DAIRY                           305
INFANT CONSUMABLE HARDLINES 

In [74]:
df[df["TripType"] == 42].Weekday.value_counts()

Sunday       4183
Saturday     4033
Friday       2910
Thursday     2164
Wednesday    2161
Monday       2036
Tuesday      1981
Name: Weekday, dtype: int64

In [105]:
df[df["TripType"] == 42].FinelineNumber.value_counts()

1805.0    152
135.0     145
808.0     128
302.0     118
0.0       114
9546.0    110
801.0     101
115.0      83
5501.0     79
1508.0     75
787.0      73
202.0      72
100.0      62
8904.0     62
6.0        59
110.0      53
3467.0     52
250.0      52
804.0      51
10.0       49
203.0      47
22.0       47
70.0       46
1101.0     45
8007.0     43
9625.0     43
324.0      42
31.0       42
403.0      42
338.0      42
         ... 
126.0       1
2010.0      1
5633.0      1
8161.0      1
5429.0      1
7892.0      1
6070.0      1
4709.0      1
8603.0      1
4290.0      1
2254.0      1
3026.0      1
4019.0      1
8325.0      1
7652.0      1
7419.0      1
1172.0      1
557.0       1
2052.0      1
168.0       1
5541.0      1
581.0       1
4715.0      1
4070.0      1
7706.0      1
814.0       1
5920.0      1
488.0       1
764.0       1
5370.0      1
Name: FinelineNumber, Length: 3336, dtype: int64

In [106]:
df[df["TripType"] == 42].ScanCount.value_counts()

 1     17417
 2      1413
 3       250
-1       176
 4       113
 5        38
 6        20
-2        14
 7         7
 8         6
 9         4
-3         4
 10        2
 11        1
-5         1
 12        1
 17        1
Name: ScanCount, dtype: int64

### TripType == 43 - One More....
- PERSONAL CARE, DSD GROCERY, IMPULSE MERCHANDISE (3개가 400을 넘음)
- 0 (거의 의미x)

In [75]:
df[df["TripType"] == 43].DepartmentDescription.value_counts()

PERSONAL CARE                  523
DSD GROCERY                    475
IMPULSE MERCHANDISE            468
PHARMACY OTC                   314
GROCERY DRY GOODS              280
OFFICE SUPPLIES                256
BEAUTY                         233
CELEBRATION                    223
HOUSEHOLD CHEMICALS/SUPP       215
PRODUCE                        208
FABRICS AND CRAFTS             166
DAIRY                          147
COOK AND DINE                  144
MENS WEAR                      133
AUTOMOTIVE                     129
HOME MANAGEMENT                128
FROZEN FOODS                   119
CANDY, TOBACCO, COOKIES        119
TOYS                           118
HOME DECOR                     114
HARDWARE                       113
SHOES                          110
LAWN AND GARDEN                103
LADIESWEAR                     102
HOUSEHOLD PAPER GOODS          101
BATH AND SHOWER                101
SPORTING GOODS                 100
FINANCIAL SERVICES              92
INFANT APPAREL      

In [76]:
df[df["TripType"] == 43].Weekday.value_counts()

Saturday     1216
Sunday       1161
Friday       1125
Monday        834
Wednesday     783
Tuesday       640
Thursday      624
Name: Weekday, dtype: int64

In [107]:
df[df["TripType"] == 43].FinelineNumber.value_counts()

0.0       92
135.0     63
5501.0    55
1805.0    52
808.0     45
100.0     34
115.0     32
9546.0    31
202.0     31
4010.0    31
1505.0    30
302.0     30
801.0     27
203.0     26
250.0     24
1508.0    23
8103.0    22
338.0     22
8945.0    21
5017.0    20
804.0     20
3467.0    20
110.0     20
324.0     20
403.0     18
787.0     17
707.0     17
3502.0    16
145.0     15
6600.0    15
          ..
1010.0     1
1532.0     1
817.0      1
3018.0     1
309.0      1
3640.0     1
8722.0     1
3457.0     1
5301.0     1
9140.0     1
4325.0     1
1209.0     1
3175.0     1
6011.0     1
8303.0     1
5107.0     1
7809.0     1
7240.0     1
5304.0     1
3617.0     1
6632.0     1
3251.0     1
1423.0     1
2802.0     1
4652.0     1
2251.0     1
8275.0     1
5039.0     1
4034.0     1
1000.0     1
Name: FinelineNumber, Length: 2004, dtype: int64

In [108]:
df[df["TripType"] == 43].ScanCount.value_counts()

 1    5784
 2     434
 3      62
-1      53
 4      28
 5       8
 6       5
-2       4
 9       4
 7       1
Name: ScanCount, dtype: int64

### TripType == 44
- PERSONAL CARE, DSD GROCERY, GROCERY DRY GOODS (첫번째께 1857개로 제일많지만 나머지것들도 1000개가 넘음)
- 0 (의미x)

In [77]:
df[df["TripType"] == 44].DepartmentDescription.value_counts()

PERSONAL CARE                  1857
DSD GROCERY                    1341
GROCERY DRY GOODS              1011
HOUSEHOLD CHEMICALS/SUPP        900
PHARMACY OTC                    889
IMPULSE MERCHANDISE             841
BEAUTY                          834
DAIRY                           685
PRODUCE                         665
CELEBRATION                     620
MENS WEAR                       567
HOUSEHOLD PAPER GOODS           512
COOK AND DINE                   507
OFFICE SUPPLIES                 493
LADIESWEAR                      462
GIRLS WEAR, 4-6X  AND 7-14      457
INFANT APPAREL                  454
FABRICS AND CRAFTS              417
SHOES                           416
PETS AND SUPPLIES               381
FROZEN FOODS                    369
BATH AND SHOWER                 363
HOME MANAGEMENT                 358
LAWN AND GARDEN                 348
INFANT CONSUMABLE HARDLINES     322
TOYS                            316
HARDWARE                        285
BOYS WEAR                   

In [78]:
df[df["TripType"] == 44].Weekday.value_counts()

Saturday     4474
Sunday       4230
Friday       2902
Wednesday    2553
Monday       2239
Thursday     2150
Tuesday      1876
Name: Weekday, dtype: int64

In [109]:
df[df["TripType"] == 44].FinelineNumber.value_counts()

0.0       127
5501.0    123
135.0     118
203.0     106
801.0      98
1508.0     98
808.0      88
1805.0     88
9546.0     79
202.0      73
115.0      72
4010.0     71
4606.0     64
8945.0     63
8101.0     62
302.0      56
100.0      55
1505.0     54
3100.0     52
110.0      51
7010.0     50
3004.0     49
6.0        48
3601.0     48
403.0      47
3006.0     46
1555.0     46
1443.0     44
6080.0     43
3658.0     43
         ... 
6204.0      1
8071.0      1
8640.0      1
106.0       1
4443.0      1
190.0       1
4122.0      1
5965.0      1
6313.0      1
26.0        1
5632.0      1
3393.0      1
8267.0      1
6582.0      1
4911.0      1
881.0       1
2608.0      1
3059.0      1
2732.0      1
3781.0      1
5571.0      1
6367.0      1
3608.0      1
7014.0      1
5613.0      1
7031.0      1
5564.0      1
763.0       1
1613.0      1
5134.0      1
Name: FinelineNumber, Length: 3275, dtype: int64

In [110]:
df[df["TripType"] == 44].ScanCount.value_counts()

 1     18113
 2      1623
 3       313
-1       139
 4       129
 5        47
 6        26
 8         6
 7         6
-2         6
 10        5
 12        4
 14        2
 9         2
 11        1
-5         1
-3         1
Name: ScanCount, dtype: int64

### TripType == 999
- FINANCIAL SERVICES, IMPULSE MERCHANDISE (2289개, 1147개로 두개가 가장많음)
- 279 (others 니깐 이것을 더세부적으로 나누지 않았을까)

In [79]:
df[df["TripType"] == 999].DepartmentDescription.value_counts()

FINANCIAL SERVICES             2289
IMPULSE MERCHANDISE            1147
LADIESWEAR                      860
MENS WEAR                       701
PERSONAL CARE                   657
DAIRY                           583
ELECTRONICS                     577
DSD GROCERY                     556
BEAUTY                          542
PHARMACY OTC                    512
HOME MANAGEMENT                 506
GROCERY DRY GOODS               501
PRODUCE                         446
AUTOMOTIVE                      364
COOK AND DINE                   358
SPORTING GOODS                  357
HARDWARE                        355
TOYS                            323
INFANT CONSUMABLE HARDLINES     301
SHOES                           296
HOME DECOR                      296
HOUSEHOLD CHEMICALS/SUPP        251
CELEBRATION                     245
LAWN AND GARDEN                 227
WIRELESS                        222
OFFICE SUPPLIES                 202
BEDDING                         196
FROZEN FOODS                

In [80]:
df[df["TripType"] == 999].Weekday.value_counts()

Saturday     2946
Sunday       2939
Friday       2756
Monday       2350
Tuesday      2236
Wednesday    2224
Thursday     2139
Name: Weekday, dtype: int64

In [113]:
df[df["TripType"] == 999].FinelineNumber.value_counts()

279.0     858
8023.0    572
1000.0    230
0.0       209
284.0     175
285.0     173
281.0     170
1505.0    170
280.0     169
1001.0    167
1508.0     80
276.0      76
275.0      74
286.0      72
5501.0     68
6.0        66
4110.0     65
1511.0     61
4635.0     59
287.0      55
808.0      55
3903.0     54
7118.0     51
135.0      48
9100.0     45
1265.0     45
2001.0     45
302.0      44
86.0       41
2010.0     39
         ... 
6612.0      1
9012.0      1
5051.0      1
7781.0      1
4653.0      1
1754.0      1
3178.0      1
6417.0      1
3338.0      1
2499.0      1
7662.0      1
2840.0      1
181.0       1
5575.0      1
1807.0      1
7274.0      1
8131.0      1
5019.0      1
9540.0      1
6245.0      1
8006.0      1
6824.0      1
8265.0      1
6520.0      1
2098.0      1
6923.0      1
1810.0      1
8514.0      1
3746.0      1
4141.0      1
Name: FinelineNumber, Length: 2919, dtype: int64

In [112]:
df[df["TripType"] == 999].ScanCount.value_counts()

-1     9131
 1     7074
 2      553
-2      493
 3       90
-3       85
-4       54
 4       52
 5       13
-5       11
 6       10
-6        8
 8        4
 7        4
-10       1
-7        1
 12       1
 13       1
 18       1
 14       1
 17       1
 15       1
Name: ScanCount, dtype: int64